### PR TITLE
fix(workflows): mutation check did not consider unstaged changes

### PR DESCRIPTION
**Pull request by eladb**
We accidentally used `--staged` in the `git diff` call that checked for mutations without calling `git add .` before.

This change consolidates the mutation check into the patch upload "action" so this is done in a single place.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -73,7 +70,7 @@ jobs:
         run: '[ -s ${{ runner.temp }}/.repo.patch ] && git apply ${{ runner.temp
           }}/.repo.patch || echo "Empty patch. Skipping."'
       - name: Found diff after build (update your branch)
-        run: git diff --staged --exit-code
+        run: git diff --exit-code
```
Comment (iliapolo) [inline .github/workflows/build.yml:73]: Applying the patch doesn't do `git add`, I think instead we just need to `git add .` - no?

Comment (iliapolo) [reply .github/workflows/build.yml:73]: @eladb The self mutation reverted this commit

File: src/build/build-workflow.ts [hunk 2]
```diff
@@ -235,7 +235,7 @@ export class BuildWorkflow extends Component {
         }),
         {
           name: 'Found diff after build (update your branch)',
-          run: 'git diff --staged --exit-code',
+          run: 'git diff --exit-code',
```
Comment (eladb) [inline src/build/build-workflow.ts:238]:
```suggestion
          run: 'git add . && diff --staged --exit-code',
```

File: src/release/release.ts [hunk 3]
```diff
@@ -457,7 +457,7 @@ export class Release extends Component {
 
     // anti-tamper check (fails if there were changes to committed files)
     // this will identify any non-committed files generated during build (e.g. test snapshots)
-    releaseTask.exec('git diff --staged --ignore-space-at-eol --exit-code');
+    releaseTask.exec('git diff --ignore-space-at-eol --exit-code');
```
Comment (iliapolo) [inline src/release/release.ts:460]: Should also have a `git add .`, which makes really uncool to run locally :\ .

Im ok addressing this later though.

**Review by iliapolo — APPROVED** [review]

**Pull request merged**